components/table/table-wrapper.js

```diff
@@ -392,6 +396,12 @@ export class TableWrapper extends RtlMixin(PageableMixin(SelectionMixin(LitEleme
 		});
 	}
 
+	_checkSelectIconButton(r) {
+		const header = Array.from(r.childNodes).find((element) => element.localName === 'th');
+		const selectAllButton = Array.from(header.childNodes).find((element) => element.localName === 'd2l-selection-select-all');
+		r.classList.toggle('d2l-table-select-icon', selectAllButton);
+	}
+
```

Comment:
Since this is a separate story/change, you'll likely want to make it in a separate PR for potentially at least a couple reasons:
- If there are issues with it but not with the other work, it ends up blocking
- If both changes result in visual diff changes, we can't isolate which change caused the change and if the changes that change specifically caused were what we intended

Related to this change:
- You should look into if we can just handle this with negative margins instead rather than adding another javascript loop-through
- I'd recommend thinking through the other cases with table that would also need this change